src/app/content/highlights/reducer.ts

```diff
@@ -38,63 +38,82 @@ const reducer: Reducer<State, AnyAction> = (state = initialState, action) => {
       };
     }
     case getType(actions.createHighlight): {
-      const newState = {...state};
-
       const highlight = {
         ...action.payload,
         color: action.payload.color as string as HighlightColorEnum,
         sourceType: action.payload.sourceType as string as HighlightSourceTypeEnum,
       };
-      newState.highlights = [...state.highlights || [], highlight];
 
-      if (newState.summary.filters.colors.includes(highlight.color)) {
-        newState.summary.highlights = addSummaryHighlight(newState.summary.highlights, {
+      let newSummaryHighlights = {};
```

Comment:
you could leave this initialized to undefined and the below logic would be simpler